### ISSUE TITLE
Generate wrong TOC

### ISSUE THREAD

**Issue by homezonebenny** [AlanWalk/Markdown-TOC#89]
**EDIT:**
You can fix it with just set it `Eol` to `\n` like descriped here:\
[GitHub issue - Solution](https://github.com/AlanWalk/markdown-toc/issues/65#issuecomment-438500104)

The **only not working thing** is that those `&nbsp;`s from the headline text goes into headline anchor link too!
So basically you have to remove them from the Anchor link manually...
This should be fixed and it's **not** a duplicate to the #65 issue post.

\
**Alternatively use the plugin called:**\
_Markdown All in One_

**After you installed this plugin you press `F1` or `CTRL + P` in your .md file and write:**\
_Create table of contents_

This will create a TOC and everytime you save the document it will be updated automatically!

**If you don't want a headline in your TOC navigation add:**\
`<!-- omit in toc -->`
directly above the headline.

For more informations read in the plugin description after the installation!

&nbsp;

----

&nbsp;

> Using VSCode Plugin **Markdown-Toc v1.5.6**

**Here's an image what it does:**
[Screenshot on Imgur](https://i.imgur.com/NdNA7xV.png)

**Basically I have normal headlines like:**
```# Main headline 1
## &nbsp;First 1
## &nbsp;Second 1
# &nbsp;Main headline 2
## &nbsp;First 2
## &nbsp;Second 2
```

**Now there are problems with it:**
- It generates a wrong TOC section. Even when I use normal headlines so your plugin is not working in basics
- If you use &nbsp; in your headline it use this the target url too and this makes a problem

**For example:**
`# 2. &nbsp;Markdown`
**Leadts to:**
`auto- [2. &nbsp;Markdown](#2-nbspmarkdown)`
**But should be:**
`- [2. &nbsp;Markdown](#25-markdown)`

This shows a correct navigation on top of the Markdown file and the target url works too.

**Comment by stoopidJSON** [comment]
I am also having this issue.

**Comment by lougreenwood** [comment]
+1

**Comment by hackjackyer** [comment]
+1

**Comment by ZakeyDrower** [comment]
+1
how to deal with it ?

**Comment by wzur-r3** [comment]
This is a duplicate of #65 - see the answer there

**Comment by hackjackyer** [comment]
> how to deal with it ?

save as \n

**Comment by mircsicz** [comment]
+1
